### PR TITLE
Support full coverage of autoscaling fields in ContainerCluster

### PR DESCRIPTION
**Pull request by maqiuyujoyce**
### Change description

Diffbase: #4713 

### Tests you have done

- [N/A] Run `make ready-pr` to ensure this PR is ready for review.
- [X] Perform necessary E2E testing for changed resources.

### CODE REVIEW

File: pkg/controller/tf/controller.go [hunk 1]
```diff
@@ -363,6 +363,12 @@ func (r *Reconciler) sync(ctx context.Context, krmResource *krmtotf.Resource) (r
 	if err != nil {
 		return false, r.HandleUpdateFailed(ctx, &krmResource.Resource, fmt.Errorf("error calculating diff: %w", err))
 	}
+	if krmResource.Kind == "ContainerCluster" && diff != nil {
+		fmt.Println("maqiuyu... containercluster...")
```
Comment (acpana) [inline pkg/controller/tf/controller.go]: remove ?

Comment (maqiuyujoyce) [reply pkg/controller/tf/controller.go]: Good catch! Will address it after #4713 is merged (as I'll need to rebase anyways).

**Review by jingyih — COMMENTED** [review]
/approve

Defer to Alex!

/hold
Feel free to unhold when PR is ready to merge.

File: pkg/controller/tf/controller.go (hunk 1 above)
Comment (maqiuyujoyce) [reply pkg/controller/tf/controller.go]: Done!

File: tests/e2e/unified_test.go [hunk 2]
```diff
@@ -378,6 +378,7 @@ func runScenario(ctx context.Context, t *testing.T, testPause bool, fixture reso
 				}
 
 				if ShouldTestRereconiliation(t, testName, primaryResource) {
+					h.Log("Testing re-reconciliation...")
```
Comment (acpana) [inline tests/e2e/unified_test.go]: nit: can we include the name of the primary resource or some other identifying string ?

File: tests/apichecks/testdata/exceptions/outputonlyspecfields.txt [hunk 3]
```diff
@@ -1,3 +1,5 @@
+[output_only_spec_field] crd=containerclusters.container.cnrm.cloud.google.com version=v1beta1: field ".spec.clusterAutoscaling.autoProvisioningDefaults.management.upgradeOptions[].autoUpgradeStartTime" is not set in unstructured objects
```
Comment (acpana) [inline tests/apichecks/testdata/exceptions/outputonlyspecfields.txt:1]: we're only adding to this file for backwards compatibility right? As in, with the changes in this PR, these fields will never get set, right?

File: pkg/test/resourcefixture/testdata/basic/container/v1beta1/containercluster/containercluster-autoscaling-surgeupgrade/update.yaml [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+      shieldedInstanceConfig:
+        enableSecureBoot: true
+        enableIntegrityMonitoring: true
+      bootDiskKMSKeyRef:
+        name: "kmscryptokey-${uniqueId}" # immutable
+      diskSize: 100 #GB
+      imageType: "UBUNTU_CONTAINERD"
+      # This field is deprecated, minCpuPlatform should be specified using
+      # cloud.google.com/requested-min-cpu-platform label selector on the pod.
+      # To unset the min cpu platform field pass "automatic" as field value.
```
Comment (acpana) [inline pkg/test/resourcefixture/testdata/basic/container/v1beta1/containercluster/containercluster-autoscaling-surgeupgrade/update.yaml:55]: wait this field is deprecated but it still takes on a value ?

File: pkg/test/resourcefixture/testdata/basic/container/v1beta1/containercluster/containercluster-autoscaling-surgeupgrade/dependencies.yaml [hunk 5]
```diff
@@ -0,0 +1,43 @@
+# Copyright 2022 Google LLC
```
Comment (acpana) [inline pkg/test/resourcefixture/testdata/basic/container/v1beta1/containercluster/containercluster-autoscaling-surgeupgrade/dependencies.yaml]:
```suggestion
# Copyright 2025 Google LLC
```

File: pkg/test/resourcefixture/testdata/basic/container/v1beta1/containercluster/containercluster-autoscaling-bluegreenupgrade/dependencies.yaml [hunk 6]
```diff
@@ -0,0 +1,43 @@
+# Copyright 2022 Google LLC
```
Comment (acpana) [inline pkg/test/resourcefixture/testdata/basic/container/v1beta1/containercluster/containercluster-autoscaling-bluegreenupgrade/dependencies.yaml]:
```suggestion
# Copyright 2025 Google LLC
```

File: mockgcp/mockcontainer/nodepool.go [hunk 7]
```diff
@@ -203,13 +203,28 @@ func (s *ClusterManagerV1) populateAutoprovisioningNodePoolDefaults(obj *pb.Auto
 			"https://www.googleapis.com/auth/cloud-platform",
 		}
 	} else {
+		// Reach out to API team to understand why when the input is
```
Comment (acpana) [inline mockgcp/mockcontainer/nodepool.go]: maybe let's create an issue and link it in here?

File: config/tests/samples/create/samples.go [hunk 8]
```diff
@@ -133,6 +133,7 @@ func RunCreateDeleteTest(t *Harness, opt CreateDeleteTestOptions) {
 				t.Fatalf("error creating resource: %v", err)
 			}
 		} else {
+			t.Log("using server-side apply to create object")
```
Comment (acpana) [inline config/tests/samples/create/samples.go]: do we need these log lines? if so, can we add the resource as the identifier?

**Review by acpana — COMMENTED** [review]
/lgtm

mostly nits! feel free to ignore/ follow up as needed

/hold

i'll let you merge it; thanks for the change 🥇

File: tests/e2e/unified_test.go (hunk 2 above)
Comment (maqiuyujoyce) [reply tests/e2e/unified_test.go]: Done.

File: tests/apichecks/testdata/exceptions/outputonlyspecfields.txt (hunk 3 above)
Comment (maqiuyujoyce) [reply tests/apichecks/testdata/exceptions/outputonlyspecfields.txt:1]: I think so. Basically, these fields were placed under spec by mistake, and cannot be covered by tests, but we must keep them under spec for backwards compatibility purposes. So I added this exception file to track such edge cases.

File: pkg/test/resourcefixture/testdata/basic/container/v1beta1/containercluster/containercluster-autoscaling-surgeupgrade/update.yaml (hunk 4 above)
Comment (maqiuyujoyce) [reply pkg/test/resourcefixture/testdata/basic/container/v1beta1/containercluster/containercluster-autoscaling-surgeupgrade/update.yaml:55]: Deprecation means a field is going to be removed eventually, but it doesn't mean the field is no longer usable now. Many fields in GKE cluster are deprecated but are kept being usable since several years ago.

File: pkg/test/resourcefixture/testdata/basic/container/v1beta1/containercluster/containercluster-autoscaling-surgeupgrade/dependencies.yaml (hunk 5 above)
Comment (maqiuyujoyce) [reply pkg/test/resourcefixture/testdata/basic/container/v1beta1/containercluster/containercluster-autoscaling-surgeupgrade/dependencies.yaml]: Done.

File: pkg/test/resourcefixture/testdata/basic/container/v1beta1/containercluster/containercluster-autoscaling-bluegreenupgrade/dependencies.yaml (hunk 6 above)
Comment (maqiuyujoyce) [reply pkg/test/resourcefixture/testdata/basic/container/v1beta1/containercluster/containercluster-autoscaling-bluegreenupgrade/dependencies.yaml]: Done.

File: mockgcp/mockcontainer/nodepool.go (hunk 7 above)
Comment (maqiuyujoyce) [reply mockgcp/mockcontainer/nodepool.go]: Created an internal ticket as we'll probably need to reach out to the API team internally, so didn't link it here.

File: config/tests/samples/create/samples.go (hunk 8 above)
Comment (maqiuyujoyce) [reply config/tests/samples/create/samples.go]: The log can be very helpful to identify whether SSA is used. Added more info as the identifier.
